### PR TITLE
90/waypoints planning

### PR DESCRIPTION
**Pull request by StefanCaldararu**
merge in waypoints path planner. currently disabled in main sim launch. works (doesn't crash), and publishes to error state.

### CODE REVIEW

File: workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        Computes the error state between the target state and the current state and the target state. First, get the target point which is closest ot the current state (plus the lookahead distance in the x direction). Then, get the error state by subtracting the current state from the target state (modular subtraction for the heading).
+
+        Returns:
+             The error state, and the target reference waypoint we are currently going to.
+
+        """
+        x_current = self.state.pose.position.x
+        y_current = self.state.pose.position.y
+        # theta_current = np.arctan2( 2*(x*w+z*y), x**2-w**2+y**2-z**2 )
+        theta_current = self.state.pose.orientation.z - 0.24845347641462115
```
Comment (AaronYoung5) [inline workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py:132]: lol long number, what is it?

File: workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                y_current + np.sin(theta_current) * self.lookahead - self.ref_traj[i][1]
+            ) ** 2
+            index = dist.argmin()
+
+        ref_state_current = list(self.ref_traj[index, :])
+        err_theta = 0
+        ref = ref_state_current[2]
+        act = theta_current
+
+        if (ref > 0 and act > 0) or (ref <= 0 and act <= 0):
```
Comment (AaronYoung5) [inline workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py:158]: not a big problem here, will approve with it, but it's probs worth describing what these lines are doing

File: workspace/src/common/launch/art_launch/launch/art.launch.py [hunk 3]
```diff
@@ -90,6 +90,12 @@ def generate_launch_description():
         "True",
         condition=IfCondition(GetLaunchArgument("use_sim")),
     )
+    SetLaunchArgument(
```
[the remark this answers is not in the document]
Comment (StefanCaldararu) [reply workspace/src/common/launch/art_launch/launch/art.launch.py:93]: idk should I just remove it from building for art_dev_meta? nothing subscribes to it in master rn

File: workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py (hunk 1 above)
Comment (StefanCaldararu) [reply workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py:132]: This has been here for a while. Weird bug that we haven't done anything about yet where the LTP we are driving on and converting GPS coordinates to is rotated by a weird amount compared to the Magnetometer readings...

Comment (StefanCaldararu) [reply workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py:132]: We should start a separate issue about that.

File: workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py (hunk 2 above)
Comment (StefanCaldararu) [reply workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py:158]: Modular subtraction (want to be able to get the difference between angles in the right direction)

Comment (StefanCaldararu) [reply workspace/src/path_planning/waypoints_path_planner/waypoints_path_planner/waypoints_path_planner.py:158]: will add a comment

**Review by AaronYoung5 — APPROVED** [review]

**Pull request merged**